🌐 Add Spanish translation for `docs/es/docs/advanced/additional-responses.md`

javillegasna opened the pull request:
Added Spanish translation to docs/es/docs/advanced/additional-responses.md

javillegasna committed:
:🌐 Add Spanish translation for docs/es/docs/advanced/additional-responses.md

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+///
+
+
+Tú puedes declarar respuestas adicionales, que contengan códigos de estado, tipos de medio, descripciones y varias cosas adicionales.
```
```suggestion
Puedes declarar respuestas adicionales, que contengan códigos de estado, tipos de medio, descripciones, etc.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Ten en cuenta que estas respuestas adicionales se incluirán en el esquema OpenAPI, por lo que también aparecerán en la documentación de la API.
+
+Pero para que esas respuestas adicionales funcionen tienes que asegurarte de que devuelves un `Response` como un objeto`JSONResponse` directamente, el cual debe incluir código de estado y contenido.
```
```suggestion
Pero para que esas respuestas adicionales funcionen, tienes que asegurarte de que devuelves un `Response` como un objeto`JSONResponse` directamente, el cual debe incluir código de estado y contenido.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Pero para que esas respuestas adicionales funcionen tienes que asegurarte de que devuelves un `Response` como un objeto`JSONResponse` directamente, el cual debe incluir código de estado y contenido.
+
+## Respuesta Adicional con un `model`
```
```suggestion
## Respuesta Adicional con `model`
```
o
```suggestion
## Respuesta Adicional usando `model`
```

javillegasna replied:
I think I'll go for the second option thx

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
````diff
@@ -0,0 +1,257 @@
+}
+```
+
+Los schemas están referenciados a otro lugar dentro del esquema de Open Api.
````
```suggestion
Los schemas están referenciados a otro lugar dentro del esquema de OpenAPI.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Puedes utilizar los mismos parámetros en tus `responses` para añadir diferentes tipos de medio en la misma respuesta principal.
+
+Por ejemplo, Puedes añadir un tipo de medio adicional de `image/png`, declarando que tu *path operation* pueda retornar un objeto JSON (con el tipo de medio `application/json`) or una imagen PNG:
```
```suggestion
Por ejemplo, Puedes añadir un tipo de medio adicional de `image/png`, declarando que tu *path operation* pueda devolver un objeto JSON (con el tipo de medio `application/json`) o una imagen PNG:
```

alejsdev committed:
Merge branch 'master' into es/advanced/additional-responses.md

javillegasna commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Este es un tema bastante avanzado.
+
+Si apenas estas empezando con **FastAPI**, Puede que no necesites esto.
```
Sounds good thanks for the suggestion

javillegasna committed:
:Apply suggestions from code review

Co-authored-by: Javier Sánchez Castro <72013291+JavierSanchezCastro@users.noreply.github.com>

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Puedes declarar un `response_model`, utilizando el código de estado `200` por defecto (o uno personalizado si lo necesitas), y después declarar información adicional para la misma respuesta en `responses`, directamente dentro del schema de OpenAPI.
+
+**FastAPI** mantendrá la información adicional desde `responses`, y la combinará con el esquema JSON desde tu modelo.
```
```suggestion
**FastAPI** mantendrá la información adicional desde `responses` y la combinará con el esquema JSON desde tu modelo.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+**FastAPI** mantendrá la información adicional desde `responses`, y la combinará con el esquema JSON desde tu modelo.
+
+Por ejemplo, puedes declarar una respuesta con un código de estado `404` que use un modelo de Pydantic y tenga un descripción personalizada.
```
```suggestion
Por ejemplo, puedes declarar una respuesta con un código de estado `404` que use un modelo de Pydantic y tenga una descripción personalizada.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
````diff
@@ -0,0 +1,257 @@
+{!../../../docs_src/additional_responses/tutorial003.py!}
+```
+
+Todo esto sera combinado e incluido dentro de tu OpenAPI, y presentado en la documentación de la API.
````
```suggestion
Todo esto será combinado e incluido dentro de tu OpenAPI, y presentado en la documentación de la API.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Es posible que desees tener algunas respuestas predefinidas que se apliquen a varios *path operations*, pero quieras combinarlas con respuestas personalizadas que sean necesarias para cada *path operation*.
+
+Para estos casos, puedes usar la técnica de python de "desempaquetar" un  `dict` con `**dict_to_unpack`:
```
```suggestion
Para estos casos, puedes usar la técnica de Python de "desempaquetar" un `dict` con `**dict_to_unpack`:
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
````diff
@@ -0,0 +1,257 @@
+}
+```
+
+Puedes usar esta técnica para rehusar algunas respuestas predefinidas en tus *path operations* y combinarlas con algunas personalizadas.
````
```suggestion
Puedes usar esta técnica para reutilizar algunas respuestas predefinidas en tus *path operations* y combinarlas con algunas personalizadas.
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
````diff
@@ -0,0 +1,257 @@
+{!../../../docs_src/additional_responses/tutorial004.py!}
+```
+
+## Mas información sobre las respuestas de OpenAPI
````
```suggestion
## Más información sobre las respuestas de OpenAPI
```

JavierSanchezCastro commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+## Mas información sobre las respuestas de OpenAPI
+
+Para mirar que exactamente puedes incluir en las respuestas, puedes chequear estas secciones dentro de la especificación de OpenAPI:
```
```suggestion
Para ver exactamente qué puedes incluir en las respuestas, puedes consultar estas secciones dentro de la especificación de OpenAPI:
```

javillegasna commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Cada uno de estos `dict`s soporta una key `model`, la cual contenga un modelo de Pydantic, de forma similar al parámetro `response_model`.
+
+**FastAPI** utilizará ese modelo, para generar su propio JSON Schema y lo incluirá en el lugar correcto dentro de OpenAPI.
```
mmm I think the first option is better understood

javillegasna commented on docs/es/docs/advanced/additional-responses.md:
```diff
@@ -0,0 +1,257 @@
+
+Amenos que especifiques explícitamente en tu parámetro `responses` un tipo de medio diferente, FastAPI asumirá que la respuesta tiene el mismo tipo de medio que las clase de respuesta principal (por defecto `application/json`).
+
+Pero si as especificado una clase de respuesta personalizada with `None` como su tipo de medio, FastAPI usará `application/json` para cualquier respuesta adicional que tenga un modelo asociado.
```
sorry I missed that word

javillegasna committed:
Apply suggestions from second code review

Correct punctuation, accent and replace some words to improve the understanding of the text.

Co-authored-by: Javier Sánchez Castro <72013291+JavierSanchezCastro@users.noreply.github.com>

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

javillegasna commented:
Hi @alejsdev and @JavierSanchezCastro  It looks like the check-labels job fails, maybe we need to remove **lang-all** label or replace on  .github/workflows/labeler.yml **one_of** check by **any_of**

JavierSanchezCastro commented:
> Hi @alejsdev and @JavierSanchezCastro It looks like the check-labels job fails, maybe we need to remove **lang-all** label or replace on .github/workflows/labeler.yml **one_of** check by **any_of**

I think it's because @github-actions has added the “docs” tag. I guess if @alejsdev removes that tag, the test will pass correctly.

alejsdev commented:
It's been fixed! Thanks to both @javillegasna @JavierSanchezCastro

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

javillegasna commented:
Maybe there is some update on this?

tiangolo committed:
Merge branch 'master' into es/advanced/additional-responses.md

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

javillegasna committed:
Merge branch 'master' into es/advanced/additional-responses.md

JavierSanchezCastro commented:
@javillegasna For my side I approved this PR time ago. You'll need another one to merge.

tiangolo commented:
This was covered in another PR, so I'll close this one now, thanks for the effort! :cake: :coffee:
